### PR TITLE
exercise by Maziar

### PR DESCRIPTION
**Pull request by Maziarmajd**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Maziar Zahedi Majd
- Your City: Bolton
- Your Slack Name: maziarmajd

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Review by humailhasankhan — COMMENTED** [review]
Nicely done Maziar. Your work looks really nice. I was just wondering about daniel's role. If Daniel was a student, would your if/else statement still return something? For instance, if you change danielsRole = "student". Other than that, I really like your answers. They are clear and understandable. I just read through your code on purpose so I do not have to go and look at the instructions. Your code clearly lets us know what it is doing.

**Comment by Maziarmajd** [comment]
> Nicely done Maziar. Your work looks really nice. I was just wondering about daniel's role. If Daniel was a student, would your if/else statement still return something? For instance, if you change danielsRole = "student". Other than that, I really like your answers. They are clear and understandable. I just read through your code on purpose so I do not have to go and look at the instructions. Your code clearly lets us know what it is doing.

Thank you Humail for your kind words. I can see your point, actually when I saw your code, ask myself same question :))

File: mandatory/2-function-creation.js [hunk 1]
```diff
@@ -6,7 +6,13 @@ Complete the function to check if the variable `num` satisfies the following req
...
 */
 
-function validate(num) {}
+function validate(num) {
+
+  isNumber = typeof num === "number";
+  isEven = num % 2 === 0;
+  isLess100 = num <= 100;
+  return isNumber && isEven && isLess100;
+}
```
Comment (Neil0696) [inline mandatory/2-function-creation.js:16]: Function is easy to read, well done!

**Comment by SteveLeicester** [comment]
Good work Maziar - thanks for this.  You have done a great job completing the exercises and the mandatory challenges.  Oniel also noted that the one liner

`stringArray.push(arrayOfStrings[i].trim().replace("/", "").toLowerCase());`

is nicely done.  This is method chaining or fluent coding style.  Its nice and concise and helps with readability.

Some things to do - to help improve coding style.  Avoid doing things that are not necessary.

So for example in your mood function (1-fix-function.js) you have

`function mood(todayMood) {`
`let isHappy = todayMood;  `
``
`  if (isHappy) {`
 `   return "I am happy";`
 ` } else {`
 `   return "I am not happy";`
 ` }`
`}`

Here you create a new local variable. It is simpler to just do

`function mood(isHappy) {`


But note that sometimes there is good reason to make a local copy of the parameter (if you want to change the value but still have access to the original for example)


Another comment I would make is about naming of variables.   When something is named isHappy it sounds like the value of the variable can be True or False that is good.  It helps the human reader (the computer doesn't mind either way).

Now take the next function

`function greaterThan10(num) {`
`  let isBigEnough= num;`
``
`  if (isBigEnough > 10) {`
`    return "num is greater than 10"; `
`  } else {`
`    return "num is not big enough";`
`  }`
`}`

So now isBigEnough also sounds like it should be True or False but it is assigned a numeric value

`  let isBigEnough= num;`

It is more consistent to do something like

`  let isBigEnough= num > 10; /* isBigEnough can be True or False Remember num > 10 is a Boolean expression */`  
`  if (isBigEnough)  { /* Test the Boolean value */`


This is a strong piece of work and shows me that you should try to find time to stretch yourself with extra challenges.  Keep it up Maziar.

Steve

**Comment by Maziarmajd** [comment]
@SteveLeicester Thank you for your kind words and great guides. there are very helpful.

**Pull request closed without merging**